src/api/firebase.js

```diff
@@ -179,21 +181,63 @@ export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 		dateCreated: new Date(),
 		// NOTE: This is null because the item has just been created.
 		// We'll use updateItem to put a Date here when the item is purchased!
-		dateLastPurchased: null,
+		dateLastPurchased: [new Date()],
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
 	});
+	console.log(newItem);
 	return newItem;
 }
 
+//Helper function to handle estimation logic
+async function handleCalculateEstimate(listRef) {
+	const selectedItem = await getDoc(listRef);
+
+	//access array of dateLastPurchased for the selected item
+	const selectedLastPurchase = selectedItem.data().dateLastPurchased;
+
+	//retrieve the most recent purchase date from the array and convert to a date
+	const lastPurchased =
+		selectedLastPurchase[selectedLastPurchase.length - 1].toDate();
+
+	//calculates the previously estimated days until next purchase
+	const previousEstimate = getDaysBetweenDates(
+		lastPurchased,
+		selectedItem.data().dateNextPurchased.toDate(),
+	);
+	let daysSincePrevPurchase = getDaysBetweenDates(lastPurchased, new Date());
+
+	//calculates a new estimate based on the previous estimate, days since last purchased, and total purchases
+	let newEstimate = calculateEstimate(
```

Comment:
refactor: I don't think we need `let` here. if the variable will not update, it's better to use `const`.